router/batchrouter/asyncdestinationmanager/salesforce-bulk-upload/api_service.go

```diff
@@ -0,0 +1,277 @@
+		return apiErr
+	}
+
+	s.logger.Debugn("Uploaded data to Salesforce Bulk job",
+		logger.NewStringField("jobID", jobID),
+	)
+
+	return nil
+}
+
+func (s *apiService) CloseJob(jobID string) *APIError {
+	reqBody := map[string]string{"state": "UploadComplete"}
+	body, _ := jsonrs.Marshal(reqBody)
+
+	endpoint := API_BASE_URL + "/jobs/ingest/" + jobID
+
+	_, apiErr := s.makeRequest(http.MethodPatch, endpoint, bytes.NewReader(body), "application/json")
+	if apiErr != nil {
+		return apiErr
+	}
+
+	s.logger.Debugn("Closed Salesforce Bulk job",
+		logger.NewStringField("jobID", jobID),
+	)
+
+	return nil
+}
+
+func (s *apiService) GetJobStatus(jobID string) (*JobResponse, *APIError) {
+	endpoint := API_BASE_URL + "/jobs/ingest/" + jobID
+
+	respBody, apiErr := s.makeRequest(http.MethodGet, endpoint, bytes.NewReader([]byte{}), "")
+	if apiErr != nil {
+		return nil, apiErr
+	}
+
+	var jobResp JobResponse
+	if err := jsonrs.Unmarshal(respBody, &jobResp); err != nil {
+		return nil, &APIError{
+			StatusCode: http.StatusInternalServerError,
+			Message:    fmt.Sprintf("unmarshalling job status: %v", err),
+			Category:   "ServerError",
+		}
+	}
+
+	return &jobResp, nil
+}
+
+func (s *apiService) GetFailedRecords(jobID string) ([]map[string]string, *APIError) {
+	endpoint := API_BASE_URL + "/jobs/ingest/" + jobID + "/failedResults"
+	return s.getCSVRecords(endpoint)
+}
+
+func (s *apiService) GetSuccessfulRecords(jobID string) ([]map[string]string, *APIError) {
+	endpoint := API_BASE_URL + "/jobs/ingest/" + jobID + "/successfulResults"
+	return s.getCSVRecords(endpoint)
+}
+
+func (s *apiService) DeleteJob(jobID string) *APIError {
+	endpoint := API_BASE_URL + "/jobs/ingest/" + jobID
+
+	_, apiErr := s.makeRequest(http.MethodDelete, endpoint, bytes.NewReader([]byte{}), "")
+	return apiErr
+}
+
+func (s *apiService) getCSVRecords(endpoint string) ([]map[string]string, *APIError) {
+	respBody, apiErr := s.makeRequest(http.MethodGet, endpoint, bytes.NewReader([]byte{}), "")
+	if apiErr != nil {
+		return nil, apiErr
+	}
+
+	reader := csv.NewReader(bytes.NewReader(respBody))
+
+	headers, err := reader.Read()
+	if errors.Is(err, io.EOF) {
+		return nil, nil
+	}
+	if err != nil {
+		return nil, &APIError{
+			StatusCode: http.StatusInternalServerError,
+			Message:    fmt.Sprintf("reading CSV header: %v", err),
+			Category:   "ServerError",
+		}
+	}
+
+	var records []map[string]string
+	for {
+		row, err := reader.Read()
+		if err != nil {
+			if errors.Is(err, io.EOF) {
+				break
+			}
+		}
+
+		record := make(map[string]string, len(headers))
+		for i, value := range row {
+			record[headers[i]] = value
+		}
+		records = append(records, record)
+	}
+
+	return records, nil
+}
+
+func (s *apiService) makeRequest(
+	method, endpoint string,
+	body io.Reader,
+	contentType string,
+) ([]byte, *APIError) {
+	respBody, apiError := s.attemptRequest(method, endpoint, body, contentType)
+	if apiError == nil {
+		return respBody, nil
+	}
+
+	return nil, apiError
+}
```

Comment:
This function is not doing anything and just proxying without any business logic. Should we remove it and dirrectly call `attemptRequest` Or rename it to `makeRequest`?